### ISSUE TITLE
[BUG] Getting error when training an XLNet model both with categ and continuous list columns

### ISSUE THREAD

**Issue by rnyak** [NVIDIA-Merlin/models#953]
### Bug description
I am getting the error below when I want to train an XLNet model both with categorical and continuous list features:

Exception: in user code:

    File "/usr/local/lib/python3.8/dist-packages/keras/engine/training.py", line 1160, in train_function  *
        return step_function(self, iterator)
    File "/usr/local/lib/python3.8/dist-packages/keras/engine/training.py", line 1146, in step_function  **
        outputs = model.distribute_strategy.run(run_step, args=(data,))
    File "/usr/local/lib/python3.8/dist-packages/keras/engine/training.py", line 1135, in run_step  **
        outputs = model.train_step(data)
    File "/usr/local/lib/python3.8/dist-packages/merlin/models/tf/models/base.py", line 904, in train_step
        outputs = self.call_train_test(x, y, sample_weight=sample_weight, training=True)
    File "/usr/local/lib/python3.8/dist-packages/merlin/models/tf/models/base.py", line 746, in call_train_test
        forward = self(
    File "/usr/local/lib/python3.8/dist-packages/keras/utils/traceback_utils.py", line 70, in error_handler
        raise e.with_traceback(filtered_tb) from None
    File "/tmp/__autograph_generated_filezjpwhizs.py", line 42, in tf__call
        ag__.for_stmt(ag__.ld(self).blocks, None, loop_body, get_state_1, set_state_1, ('context', 'outputs'), {'iterate_names': 'block'})
    File "/tmp/__autograph_generated_filezjpwhizs.py", line 40, in loop_body
        (outputs, context) = ag__.converted_call(ag__.ld(self)._call_child, (ag__.ld(block), ag__.ld(outputs), ag__.ld(context)), None, fscope)
    File "/tmp/__autograph_generated_file5usxendk.py", line 25, in tf___call_child
        outputs = ag__.converted_call(ag__.ld(call_layer), (ag__.ld(child), ag__.ld(inputs)), dict(**ag__.ld(call_kwargs)), fscope)
    File "/tmp/__autograph_generated_filekejmxhfy.py", line 50, in tf__call_layer
        retval_ = ag__.converted_call(ag__.ld(layer), ((ag__.ld(inputs),) + tuple(ag__.ld(args))), dict(**ag__.ld(filtered_kwargs)), fscope)
    File "/tmp/__autograph_generated_file1zh0y0oh.py", line 28, in tf___tabular_call
        ag__.if_stmt(ag__.converted_call(ag__.ld(isinstance), (ag__.ld(outputs), ag__.ld(dict)), None, fscope), if_body, else_body, get_state, set_state, ('outputs',), 1)
    File "/tmp/__autograph_generated_file1zh0y0oh.py", line 23, in if_body
        outputs = ag__.converted_call(ag__.ld(self).post_call, (ag__.ld(outputs),), dict(transformations=ag__.ld(post), merge_with=ag__.ld(merge_with), aggregation=ag__.ld(aggregation)), fscope)
    File "/tmp/__autograph_generated_fileafna7rjh.py", line 116, in tf__post_call
        ag__.if_stmt(ag__.ld(_aggregation), if_body_4, else_body_4, get_state_5, set_state_5, ('do_return', 'retval_'), 2)
    File "/tmp/__autograph_generated_fileafna7rjh.py", line 103, in if_body_4
        retval_ = ag__.converted_call(ag__.ld(_aggregation), (ag__.ld(outputs),), None, fscope)
    File "/tmp/__autograph_generated_filebeohsfqu.py", line 14, in tf____call__
        retval_ = ag__.converted_call(ag__.converted_call(ag__.ld(super), (), None, fscope).__call__, tuple(ag__.ld(args)), dict(**ag__.ld(kwargs)), fscope)
    File "/tmp/__autograph_generated_filexbhykeg4.py", line 10, in tf__call
        ag__.converted_call(ag__.ld(self)._check_concat_shapes, (ag__.ld(inputs),), None, fscope)
    File "/tmp/__autograph_generated_file7iikmetp.py", line 21, in tf___check_concat_shapes
        ag__.if_stmt((ag__.converted_call(ag__.ld(len), (ag__.converted_call(ag__.ld(set), ([ag__.converted_call(ag__.ld(tuple), (ag__.ld(v)[:(- 1)],), None, fscope) for v in ag__.converted_call(ag__.ld(input_sizes).values, (), None, fscope)],), None, fscope),), None, fscope) > 1), if_body, else_body, get_state, set_state, (), 0)
    File "/tmp/__autograph_generated_file7iikmetp.py", line 17, in if_body
        raise ag__.converted_call(ag__.ld(Exception), (ag__.converted_call('All features dimensions except the last one must match: {}'.format, (ag__.ld(input_sizes),), None, fscope),), None, fscope)

    Exception: Exception encountered when calling layer "model" "                 f"(type Model).
    
    in user code:
    
        File "/usr/local/lib/python3.8/dist-packages/merlin/models/tf/models/base.py", line 1468, in call  *
            outputs, context = self._call_child(block, outputs, context)
        File "/usr/local/lib/python3.8/dist-packages/merlin/models/tf/models/base.py", line 1497, in _call_child  *
            outputs = call_layer(child, inputs, **call_kwargs)
        File "/usr/local/lib/python3.8/dist-packages/merlin/models/tf/utils/tf_utils.py", line 437, in call_layer  *
            return layer(inputs, *args, **filtered_kwargs)
        File "/usr/local/lib/python3.8/dist-packages/merlin/models/tf/core/tabular.py", line 481, in _tabular_call  *
            outputs = self.post_call(
        File "/usr/local/lib/python3.8/dist-packages/merlin/models/tf/core/tabular.py", line 264, in post_call  *
            return _aggregation(outputs)
        File "/usr/local/lib/python3.8/dist-packages/merlin/models/config/schema.py", line 58, in __call__  *
            return super().__call__(*args, **kwargs)
        File "/usr/local/lib/python3.8/dist-packages/keras/utils/traceback_utils.py", line 70, in error_handler  **
            raise e.with_traceback(filtered_tb) from None
        File "/tmp/__autograph_generated_filexbhykeg4.py", line 10, in tf__call
            ag__.converted_call(ag__.ld(self)._check_concat_shapes, (ag__.ld(inputs),), None, fscope)
        File "/tmp/__autograph_generated_file7iikmetp.py", line 21, in tf___check_concat_shapes
            ag__.if_stmt((ag__.converted_call(ag__.ld(len), (ag__.converted_call(ag__.ld(set), ([ag__.converted_call(ag__.ld(tuple), (ag__.ld(v)[:(- 1)],), None, fscope) for v in ag__.converted_call(ag__.ld(input_sizes).values, (), None, fscope)],), None, fscope),), None, fscope) > 1), if_body, else_body, get_state, set_state, (), 0)
        File "/tmp/__autograph_generated_file7iikmetp.py", line 17, in if_body
            raise ag__.converted_call(ag__.ld(Exception), (ag__.converted_call('All features dimensions except the last one must match: {}'.format, (ag__.ld(input_sizes),), None, fscope),), None, fscope)
    
        Exception: Exception encountered when calling layer "concat_features" "                 f"(type ConcatFeatures).
        
        in user code:
        
            File "/usr/local/lib/python3.8/dist-packages/merlin/models/tf/core/aggregation.py", line 56, in call  *
                self._check_concat_shapes(inputs)
            File "/usr/local/lib/python3.8/dist-packages/merlin/models/tf/core/tabular.py", line 52, in _check_concat_shapes  *
                raise Exception(
        
            Exception: All features dimensions except the last one must match: {'item_id_seq': TensorShape([None, None, 32]), 'categories': TensorShape([None, None, 16]), 'item_age_days_norm': TensorShape([None, None]), 'event_hour_sin': TensorShape([None, None]), 'event_hour_cos': TensorShape([None, None]), 'event_weekday_sin': TensorShape([None, None]), 'event_weekday_cos': TensorShape([None, None])}
        
        
        Call arguments received by layer "concat_features" "                 f"(type ConcatFeatures):
          • inputs={'item_id_seq': 'tf.RaggedTensor(values=Tensor("model/parallel_block/embeddings/item_id_seq/item_id_seq/embedding_lookup_ragged/embedding_lookup/Identity_1:0", shape=(None, 32), dtype=float32), row_splits=Tensor("sequence_mask_random/PartitionedCall:13", shape=(None,), dtype=int32))', 'categories': 'tf.RaggedTensor(values=Tensor("model/parallel_block/embeddings/categories/categories/embedding_lookup_ragged/embedding_lookup/Identity_1:0", shape=(None, 16), dtype=float32), row_splits=Tensor("sequence_mask_random/PartitionedCall:1", shape=(None,), dtype=int32))', 'item_age_days_norm': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/PartitionedCall:10", shape=(None,), dtype=float32), row_splits=Tensor("sequence_mask_random/PartitionedCall:11", shape=(None,), dtype=int32))', 'event_hour_sin': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/PartitionedCall:4", shape=(None,), dtype=float32), row_splits=Tensor("sequence_mask_random/PartitionedCall:5", shape=(None,), dtype=int32))', 'event_hour_cos': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/PartitionedCall:2", shape=(None,), dtype=float32), row_splits=Tensor("sequence_mask_random/PartitionedCall:3", shape=(None,), dtype=int32))', 'event_weekday_sin': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/PartitionedCall:8", shape=(None,), dtype=float32), row_splits=Tensor("sequence_mask_random/PartitionedCall:9", shape=(None,), dtype=int32))', 'event_weekday_cos': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/PartitionedCall:6", shape=(None,), dtype=float32), row_splits=Tensor("sequence_mask_random/PartitionedCall:7", shape=(None,), dtype=int32))'}
          • kwargs={'mask': {'item_id_seq': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/StatefulPartitionedCall:0", shape=(None,), dtype=bool), row_splits=Tensor("sequence_mask_random/StatefulPartitionedCall:1", shape=(None,), dtype=int64))', 'categories': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/StatefulPartitionedCall:0", shape=(None,), dtype=bool), row_splits=Tensor("sequence_mask_random/StatefulPartitionedCall:1", shape=(None,), dtype=int64))', 'item_age_days_norm': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/StatefulPartitionedCall:10", shape=(None,), dtype=bool), row_splits=Tensor("sequence_mask_random/StatefulPartitionedCall:11", shape=(None,), dtype=int64))', 'event_hour_sin': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/StatefulPartitionedCall:4", shape=(None,), dtype=bool), row_splits=Tensor("sequence_mask_random/StatefulPartitionedCall:5", shape=(None,), dtype=int64))', 'event_hour_cos': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/StatefulPartitionedCall:2", shape=(None,), dtype=bool), row_splits=Tensor("sequence_mask_random/StatefulPartitionedCall:3", shape=(None,), dtype=int64))', 'event_weekday_sin': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/StatefulPartitionedCall:8", shape=(None,), dtype=bool), row_splits=Tensor("sequence_mask_random/StatefulPartitionedCall:9", shape=(None,), dtype=int64))', 'event_weekday_cos': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/StatefulPartitionedCall:6", shape=(None,), dtype=bool), row_splits=Tensor("sequence_mask_random/StatefulPartitionedCall:7", shape=(None,), dtype=int64))'}, 'training': 'True'}
    
    
    Call arguments received by layer "model" "                 f"(type Model):
      • inputs={'item_age_days_norm': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/PartitionedCall:10", shape=(None,), dtype=float32), row_splits=Tensor("sequence_mask_random/PartitionedCall:11", shape=(None,), dtype=int32))', 'event_hour_sin': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/PartitionedCall:4", shape=(None,), dtype=float32), row_splits=Tensor("sequence_mask_random/PartitionedCall:5", shape=(None,), dtype=int32))', 'event_hour_cos': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/PartitionedCall:2", shape=(None,), dtype=float32), row_splits=Tensor("sequence_mask_random/PartitionedCall:3", shape=(None,), dtype=int32))', 'event_weekday_sin': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/PartitionedCall:8", shape=(None,), dtype=float32), row_splits=Tensor("sequence_mask_random/PartitionedCall:9", shape=(None,), dtype=int32))', 'event_weekday_cos': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/PartitionedCall:6", shape=(None,), dtype=float32), row_splits=Tensor("sequence_mask_random/PartitionedCall:7", shape=(None,), dtype=int32))', 'item_id_seq': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/PartitionedCall:12", shape=(None,), dtype=int64), row_splits=Tensor("sequence_mask_random/PartitionedCall:13", shape=(None,), dtype=int32))', 'categories': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/PartitionedCall:0", shape=(None,), dtype=int64), row_splits=Tensor("sequence_mask_random/PartitionedCall:1", shape=(None,), dtype=int32))'}
      • targets={'item_id_seq': 'tf.RaggedTensor(values=Tensor("sequence_mask_random/PartitionedCall:14", shape=(None,), dtype=int64), row_splits=Tensor("sequence_mask_random/PartitionedCall:15", shape=(None,), dtype=int32))'}
      • training=True
      • testing=False
      • output_context=False

```
### Steps/Code to reproduce bug

Please run this code to repro the error:


import merlin.models.tf as mm
from merlin.schema import Tags
from merlin.datasets.synthetic import generate_data
from merlin.models.tf.transforms.tensor import ListToDense, ListToRagged, ProcessList
import os 
NUM_ROWS = int(os.environ.get("NUM_ROWS", '1000'))

sequence_testing_data = generate_data('sequence-testing', int(NUM_ROWS))

sequence_testing_data.schema = sequence_testing_data.schema.select_by_tag(Tags.SEQUENCE)
seq_schema = sequence_testing_data.schema

seq_inputs = mm.InputBlockV2(
    seq_schema,
    categorical=mm.Embeddings(
        seq_schema.select_by_tag(Tags.CATEGORICAL), sequence_combiner=None
    ),
    pre=ProcessList(seq_schema)
)

dmodel=64
mlp_block = mm.MLPBlock(
                [128,dmodel],
                activation='relu',
                no_activation_last_layer=True,
            )
model = mm.Model(
    seq_inputs,
    mlp_block,
    mm.XLNetBlock(d_model=dmodel, n_head=4, n_layer=2, 
                  pre=mm.ReplaceMaskedEmbeddings(),
                  post="inference_hidden_state",
                 ),
    mm.CategoricalOutput(
        seq_schema.select_by_name(['item_id_seq']),
        default_loss="categorical_crossentropy",
    ),
)

model.compile(run_eagerly=False, optimizer='adam', loss="categorical_crossentropy")
model.fit(sequence_testing_data, batch_size=128, epochs=1, pre=mm.SequenceMaskRandom(schema=seq_schema, target='item_id_seq', masking_prob=0.3))
```

### Environment details
- Merlin version: 
- Platform: Docker 
- Python version:
- PyTorch version (GPU?):
- Tensorflow version (GPU?):

Using `merlin-tensorflow:22.12` docker image with the latest main branch pulled.

**Comment by rnyak** [comment]
fixed by https://github.com/NVIDIA-Merlin/models/pull/969
